### PR TITLE
Add Stack + Nix CI integration

### PR DESCRIPTION
**Pull request by psibi**
This PR tests against latest three stackage resolvers covering different major ghc versions.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	stack --nix test
+
+## Test for lts18
+test-lts18:
+	make shell-lts18
+	stack --nix test --resolver lts-18.0
+
+## Test for lts16
+test-lts16:
+	make shell-lts16
```
Comment (cdepillabout) [inline Makefile:30]: nitpick: I'm not very familiar with `make`, but generally when I see people using recursive `make`, they call it recursively like `$(MAKE)`: https://www.gnu.org/software/make/manual/html_node/Recursion.html.

**Review by cdepillabout — COMMENTED** [review]
review

**Review by cdepillabout — COMMENTED** [review]
review

File: Makefile (hunk 1 above)
Comment (psibi) [reply Makefile:30]: Thanks, I wasn't aware of that!

**Comment by psibi** [comment]
After speaking to @cdepillabout , we decided that it's probably not a good idea to use Nix with a CI like that.

**Comment by jappeace** [comment]
Hi, I'm a bit confused, could you clarify on why it's not a good idea to use nix? 
Can't we test against different versions of ghc with nix? I'm reasonably sure I've done that before ([here](https://github.com/SupercedeTech/haskell-xmpp/blob/master/default.nix#L5) for example)

Have a good weekend :slightly_smiling_face:

**Comment by psibi** [comment]
@jappeace There were two reasons specified by @cdepillabout  (correct me if I mis-understood something!):

* If the idea is to test against last three stackage lts resolvers, it's just much easier to test it using the usual approach of using stack. (similar to how it's done for the package fakedata)
* This package doesn't have any native dependency etc to actually make it useful to use Nix.

Also, if you can see the CI builds, the `lts14` builds fail because of some weird issue going on with the way Nix packages `ghc865Binary`. So for this one case - we have to either go to an older nix package set to make it work or actually fix the nix issue.

**Pull request closed without merging**